concrete/dense_undirected_matrix.go

```diff
@@ -0,0 +1,130 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package concrete
+
+import (
+	"github.com/gonum/graph"
+	"github.com/gonum/matrix/mat64"
+)
+
+// UndirectedDenseGraph represents a graph such that all IDs are in a contiguous
+// block from 0 to n-1.
+type UndirectedDenseGraph struct {
+	absent float64
+	mat    *mat64.SymDense
+}
+
+// NewUndirectedDenseGraph creates an undirected dense graph with n nodes.
+// If passable is true all nodes will have an edge with unit cost, otherwise
```

Comment:
Sorry for nitpicking, but this comment should be modified in line with `NewDirectedDenseGraph()`.